purchase_sale_inter_company/tests/test_inter_company_purchase_sale.py

```diff
@@ -0,0 +1,396 @@
+from odoo.tests import Form
+
+from odoo.addons.account_invoice_inter_company.tests.test_inter_company_invoice import (
+    TestAccountInvoiceInterCompanyBase,
+)
+
+
+class TestPurchaseSaleInterCompany(TestAccountInvoiceInterCompanyBase):
+    @classmethod
+    def _configure_user(cls, user):
+        for xml in [
+            "account.group_account_manager",
+            "base.group_partner_manager",
+            "sales_team.group_sale_manager",
+            "purchase.group_purchase_manager",
+        ]:
+            user.groups_id |= cls.env.ref(xml)
+
+    @classmethod
+    def _create_purchase_order(cls, partner, products=None):
+        if not products:
+            products = [None]
+
+        po = Form(cls.env["purchase.order"])
+        po.company_id = cls.company_a
+        po.partner_id = partner
+
+        cls.product.invoice_policy = "order"
+
+        for product in products:
+            with po.order_line.new() as line_form:
+                line_form.product_id = product or cls.product
+                line_form.product_qty = 3.0
+                line_form.price_unit = 450.0
+        return po.save()
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        # no job: avoid issue if account_invoice_inter_company_queued is installed
+        cls.env = cls.env(context={"test_queue_job_no_delay": 1})
+
+        cls.product = cls.product_consultant_multi_company
+        cls.service_product_2 = cls.env["product.product"].create(
+            {
+                "name": "Service Product 2",
+                "type": "service",
+            }
+        )
+        # if product_multi_company is installed
+        if "company_ids" in cls.env["product.template"]._fields:
+            # We have to do that because the default method added a company
+            cls.service_product_2.company_ids = False
+
+        if "company_ids" in cls.env["res.partner"]._fields:
+            # Intercompany contacts should not have a company set
+            cls.partner_company_a.company_ids = False
+            cls.partner_company_b.company_ids = False
+
+        # Configure Company B (the supplier)
+        cls.company_b.so_from_po = True
+        cls.company_b.sale_auto_validation = 1
+
+        cls.intercompany_sale_user_id = cls.user_company_b.copy()
+        cls.intercompany_sale_user_id.company_ids |= cls.company_a
+        cls.company_b.intercompany_sale_user_id = cls.intercompany_sale_user_id
+
+        # Configure User
+        cls._configure_user(cls.user_company_a)
+        cls._configure_user(cls.user_company_b)
+
+        # Create purchase order
+        cls.purchase_company_a = cls._create_purchase_order(cls.partner_company_b)
+
+        # Create account
+        income_account = cls.env["account.account"].create(
+            {
+                "name": "test_account_income",
+                "code": "987",
+                "account_type": "income",
+                "company_ids": [(6, 0, [cls.company_b.id])],
```

Comment:
Please use [Command](https://github.com/odoo/odoo/blob/028e7228cb830e47a9726bef4c82793ba4590cd5/odoo/fields.py#L4237-L4266
), it is more readable

```suggestion
                "company_ids": [Command.set([cls.company_b.id])],
```

Reply:
Done !